xapian-core/configure.ac

```diff
@@ -893,6 +905,7 @@ case $enable_backend_glass$enable_backend_honey in
     ])
   if test x != x"$LIBS" ; then
     XAPIAN_LIBS="$XAPIAN_LIBS $LIBS"
+    dnl CMake finds zlib itself using FindZLIB.cmake
```

Comment:
Is that really a good idea?  While zlib has very good long term ABI stability, probing for zlib separately at application build time could find a different zlib which was built in an incompatible way (e.g. some platforms have multiple incompatible ABIs; there's also situations like using memsan which requires rebuilding all libraries in use with memsan to work properly).  It seems safer to use whatever library is picked for zlib here.